spec/services/form_response_spec.rb

```diff
@@ -29,6 +29,74 @@
       end
     end
 
+    context 'initialized with hash for failure reasons' do
```

Comment:
rspec is... weird. Your code is great Ruby and a well-thought-out test, but there are a few conventions that I want to mention for consistency.

First, this is inside the `describe '#errors'` block. This is a weird case, because `errors` isn't actually a concrete method defined in FormReponse, but is rather implicitly made one through `attr_reader :errors`. But generally, there's a describe block per method. I'd move what you added into a new `describe '#failure_reasons' do` block.

(Oh also, I don't think it matters a ton in this case, but should we pass `success: false` instead of `success: true`? That feels like a more realistic scenario.)

That's probably the only stuff important to change in this file, because the rest of the file doesn't follow any of the other conventions I'm going to mention. o_O

---------

But for completeness, a couple other ideas to try on for size, but no worries if they don't feel right here. The rest of the file doesn't do these things.

Where we keep repeating stuff like:

```ruby
response = FormResponse.new(success: true, errors: errors)

expect(response.failure_reasons).to eq failure_reasons
```

It might be easier to just [define an explicit subject](https://relishapp.com/rspec/rspec-core/docs/subject/explicit-subject). You can see an example of this here:

https://github.com/18F/identity-idp/blob/main/spec/lib/telephony/test/message_spec.rb#L5

The other idiomatic rspec bit that might help is using [let](https://relishapp.com/rspec/rspec-core/v/2-11/docs/helper-methods/let-and-let) to set up variables inside `describe` or `context` blocks.

Putting the two together can be powerful. I think I'd end up writing this as something like (nesting contexts):

```ruby
  describe '#failure_reasons' do

    subject { FormResponse.new(success: false, errors: errors).failure_reasons }

    context 'initialized with hash for failure reasons' do
      context 'when the error is a string' do
        let(:errors) do
          { foo: 'bar' }
        end

        it 'returns the value of the failure_reasons' do
          expect(subject).to eq 'foo bar'
        end
      end

      context 'when the error is a symbol' do
        let(:errors) do
          { foo: :bar }
        end

        it 'returns the value of the failure_reasons' do
          expect(subject).to eq 'foo bar'
        end
...
```

(Note that things defined with `let` are lazily evaluated, as is `subject`, so it's valid to reference `errors` even though the `let(:errors)` block is defined lower inside the contexts.)

Normally `let`s are handled inline, like `let(:user_id} { 1234 }`, but in the case where it's a hash, `let(:errors) { { foo: :bar } }` looks really confusing to me so I made it a multi-line block just for readability.

All that said -- your test perfectly follows the existing pattern in that file, so don't feel like you need to do a substantial rewrite of this stuff for your PR.

(Note that the RelishApp site is the canonical rspec documentation, which I linked to for "let" and "define an explicit subject," but I have always found their examples really confusing.)